### PR TITLE
restarting when recovery instead of cleaning contexts

### PR DESCRIPTION
**Pull request by natabeck**

### CODE REVIEW

File: worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java [hunk 1]
```diff
@@ -41,7 +41,14 @@
     private volatile String wrv; //must be volatile since it is read/written in several threads
 
 	public void doRecovery(){
-		try{
+        try {
+            boolean toRestart = Boolean.getBoolean("restart.on.recovery");
+            //If we are configured to restart on recovery - do shutdown
+            if(toRestart){
+                logger.warn("Worker is configured to restart on recovery and since internal recovery is needed the process is exiting...");
+                System.exit(255);
```
Comment (meirwah) [inline worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java]: System.exit(255) ? really?

Comment (orius123) [inline worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java]: Any preferred numbers?

File: worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java [hunk 2]
```diff
@@ -41,7 +41,14 @@
     private volatile String wrv; //must be volatile since it is read/written in several threads
 
 	public void doRecovery(){
-		try{
+        try {
+            boolean toRestart = Boolean.getBoolean("restart.on.recovery");
```
Comment (meirwah) [inline worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java]: maybe a more specific env var ("cloudslang.worker.restart.on.recovery")

File: worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java [hunk 3]
```diff
@@ -23,9 +23,10 @@
  */
 public class WorkerRecoveryManagerImpl implements WorkerRecoveryManager {
 
-	protected static final Logger logger = Logger.getLogger(WorkerRecoveryManagerImpl.class);
+    protected static final Logger logger = Logger.getLogger(WorkerRecoveryManagerImpl.class);
+    private static final int EXIT_STATUS = 111;
```
Comment (orius123) [inline worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java]: @natabeck sorry for this BS but mo OCD can't give up....

We can go with 75 - this is as close to a standart as I could find.
from: [/usr/include/sysexits.h](http://www.opensource.apple.com/source/Libc/Libc-320/include/sysexits.h)
```

#define EX_TEMPFAIL	75	/* temp failure; user is invited to retry */
 *	EX_TEMPFAIL -- temporary failure, indicating something that
 *		is not really an error.  In sendmail, this means
 *		that a mailer (e.g.) could not create a connection,
 *		and the request should be reattempted later.
```

File: worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java (hunk 2 above)
Comment (orius123) [inline worker/worker-manager/score-worker-manager-impl/src/main/java/io/cloudslang/worker/management/services/WorkerRecoveryManagerImpl.java]: :+1:

**Comment by orius123** [comment]
@natabeck change to `cloudslang.worker.restart.on.recovery` and merge please

**Pull request merged**